pkg/model/hetznermodel/sshkey.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hetznermodel
+
+import (
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/hetzner"
+	"k8s.io/kops/upup/pkg/fi/cloudup/hetznertasks"
+)
+
+// SSHKeyModelBuilder configures SSHKey objects
+type SSHKeyModelBuilder struct {
+	*HetznerModelContext
+	Lifecycle fi.Lifecycle
+}
+
+var _ fi.ModelBuilder = &SSHKeyModelBuilder{}
+
+func (b *SSHKeyModelBuilder) Build(c *fi.ModelBuilderContext) error {
+	sshkey := &hetznertasks.SSHKey{
+		Name:      fi.String(b.ClusterName()),
```

Comment:
As a trick, you could also use `LinktoSSHKey().Name` here also.  That's a trick I've just started using, but I think I like it :-)